source-bamboo-hr: Add employees stream

mkrawc opened the pull request:
## What
Add new employees stream based on https://documentation.bamboohr.com/reference/get-employee-1

## How
- Add employees stream to manifest.yaml
- Employees stream should always read all rows but can be configured as incremental stream to capture employee changes over time, like salary changes. A user can define what fields should be tracked by setting a multi column primary key. 
- Refactor `CustomReportsSchemaLoader` to read fields from config and define the default schema loader defined in manifest


## User Impact
New streams are available

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
- [ ] NO ❌

marcosmarxm reviewed (changes requested):
It seems this contribution is making multiple changes at once. @mkrawc, please move the timesheet to another contribution as a simple stream addition. We can accept that faster and release a new version. The employee stream is making many changes that impact the current custom report, so we will need more discussion about the best implementation.

fumeoss commented:
## Summary
- Added new streams for employees and timesheet entries to the BambooHR connector.
- Employees stream can be configured for incremental updates to track changes over time using a multi-column primary key.
- Refactored the `CustomReportsSchemaLoader` to allow dynamic field reading from configuration and set default schema loading behavior.
- New configuration options included for specifying employee fields and tracking additional default fields.

mkrawc commented on airbyte-integrations/connectors/source-bamboo-hr/components.py:
```diff
@@ -53,94 +57,3 @@ def convert_custom_reports_fields_to_list(self, custom_reports_fields: str) -> l
         return schema1
-
-
-CUSTOM_REPORTS_BASE_SCHEMA = {
```
I moved the static schema to inline schema in the manifest.yaml and use `default_schema_loader` schema loader to load it here

mkrawc commented:
@marcosmarxm I moved the timesheet_entries stream to a separate PR: https://github.com/airbytehq/airbyte/pull/54701 and removed it from this PR. I also added some comments that explain the code and thinking behind it. Please review.

marcosmarxm reviewed (approved):
Thanks @mkrawc
